### ISSUE TITLE
private registry: curl registry works, but docker pull/login gets Client.Timeout exceeded error

### ISSUE THREAD

**Issue by rezroo** [distribution/distribution#2527]
I'm accessing a remote private registry over an openvpn container.  I just want to pull the image into the AWS instance. Everything works on the remote-registry node (pull/push/login/etc). On the AWS side I can curl the remote registry, but nothing else works. What else should I check or look at?

[ec2-user@ip-172-31-24-144 ~]$ sudo nsenter -t 23288 -n curl --insecure https://docker-registry:4000/v2/_catalog
{"repositories":["fuel100"]}

[ec2-user@ip-172-31-24-144 ~]$ sudo nsenter -t 23288 -n docker pull docker-registry:4000/fuel100:1
Error response from daemon: Get http://docker-registry:4000/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

[ec2-user@ip-172-31-24-144 ~]$ tail -f /var/log/docker
time="2018-02-18T03:09:32.967253029Z" level=warning msg="Error getting v2 registry: Get https://docker-registry:4000/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"
time="2018-02-18T03:09:32.967309688Z" level=info msg="Attempting next endpoint for pull after error: Get https://docker-registry:4000/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"
time="2018-02-18T03:09:47.967504865Z" level=warning msg="Error getting v2 registry: Get http://docker-registry:4000/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"
time="2018-02-18T03:09:47.967546177Z" level=info msg="Attempting next endpoint for pull after error: Get http://docker-registry:4000/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"
time="2018-02-18T03:09:47.967573503Z" level=error msg="Handler for POST /v1.32/images/create returned error: Get http://docker-registry:4000/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)"

[ec2-user@ip-172-31-24-144 ~]$ docker version
Client:
 Version:      17.09.1-ce
 API version:  1.32
 Go version:   go1.8.4
 Git commit:   3dfb8343b139d6342acfd9975d7f1068b5b1c3d3
 Built:        Mon Jan  8 22:44:09 2018
 OS/Arch:      linux/amd64

Server:
 Version:      17.09.1-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.4
 Git commit:   402dd4a/17.09.1-ce
 Built:        Mon Jan  8 22:44:42 2018
 OS/Arch:      linux/amd64
 Experimental: false

stack@lab:~$ docker exec privmirror registry --version
registry github.com/docker/distribution v2.6.2

**Comment by xiaods** [comment]
need add insecure-registry flag to docker daemon config.

**Comment by dmp42** [comment]
@rezroo you probably should start looking at your network traffic in more details (tcpdump).

Somehow docker is unable to communicate through your vpn. It's hard to advise on what is going on here, since this is likely specific to your config / VPN.

**Comment by Heermosi** [comment]
> @rezroo you probably should start looking at your network traffic in more details (tcpdump).
> 
> Somehow docker is unable to communicate through your vpn. It's hard to advise on what is going on here, since this is likely specific to your config / VPN.
I'm sorry, I jumped in, but, I have also faced with the same issue, and I suspect some changes to the network caused this.
But how can I prove it?(If I proved it, then I would not be blamed for that)
I used TCP dump, all packages reached the destination, except for check sum error, which can be ignored.
Exactly it seems that after receiving the https cert, the docker dropped the connection dues to time out but I don't know what is it requesting and I did not see anything meaningful sent between the client and registry.
Any details should I provide to help you analyze the problem?
